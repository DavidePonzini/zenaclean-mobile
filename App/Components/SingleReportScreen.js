import React, { Component } from 'react'
import { StyleSheet, View, Text, Image, ScrollView } from 'react-native'
import DateParser from '../Utils/DateParser'
import Images from '../Themes/Images'
import Fonts from '../Themes/Fonts'
import Metrics from '../Themes/Metrics'

export default class SingleReportScreen extends Component {
  constructor (props) {
    super(props)
    this.marker = props.navigation.state.params.marker
    this.imgUri = (this.marker.url == null) ? Images.defaultReportPicture : { uri: this.marker.url }
    const { date, time } = DateParser.timestampToItalianDate(this.marker.timestamp)
    this.date = date
    this.time = time
  }
  static navigationOptions = {
    headerStyle: {
      backgroundColor: 'dodgerblue'
    },
    headerTintColor: '#fff',
    headerTitleStyle: {
      fontWeight: 'bold'
    }
  }

  render () {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.title}>{this.marker.title}</Text>
          <Text style={styles.address}>{this.marker.address == null ? 'Indirizzo sconosciuto' : this.marker.address}, {this.date + ' alle ' + this.time}</Text>
        </View>
        <ScrollView style={styles.scrollable}>
          <Image style={styles.image} source={this.imgUri} />
          <View style={styles.innerContainer}>
            <Text style={styles.description}>{this.marker.description}</Text>
            <View style={styles.footer} />
          </View>
        </ScrollView>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#FEFEFE'
  },
  header: {
    padding: 5
  },
  scrollable: {
    flexDirection: 'column'
  },
  innerContainer: {
    padding: 15
  },
  footer: {
    flex: 0.1
  },
  image: {
    width: Metrics.screenWidth,
    height: Metrics.screenWidth
  },
  title: {
    ...Fonts.style.h3,
    borderBottomColor: '#EAEAEA',
    textAlign: 'center',
    paddingBottom: 5,
    marginBottom: 10,
    borderBottomWidth: 1
  },
  address: {
    fontSize: Fonts.size.small,
    textAlign: 'center',
    padding: 5
  },
  row: {
    flexDirection: 'row'
  }
})
