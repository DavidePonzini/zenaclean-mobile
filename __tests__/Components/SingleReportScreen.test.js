import React from 'react'
import SingleReportScreen from '../../App/Components/SingleReportScreen'

import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import 'isomorphic-fetch'

configure({ adapter: new Adapter() })

const fixtureMarker = { 'description': 'In mezzo alla strada', 'latitude': 44.400879, 'longitude': 8.972585, 'timestamp': '2018-02-28T06:35:33 -01:00', 'title': 'Divano abbandonato' }
const fixtureNavigation = { navigate: jest.fn(),
  state: {
    params: {
      marker: fixtureMarker
    }
  }
}

jest.mock('../../App/Services/ApiService', () => require('../../App/Services/FixtureApiService'))

test('renders correctly', () => {
  const tree = renderer.create(<SingleReportScreen navigation={fixtureNavigation} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('coordinates converted to address', async () => {
  const wrapper = shallow(<SingleReportScreen navigation={fixtureNavigation} />)
  const instance = wrapper.instance()
  await instance.componentWillMount()
  expect(instance.state.address).toBe('Via del Camoscio')
})
