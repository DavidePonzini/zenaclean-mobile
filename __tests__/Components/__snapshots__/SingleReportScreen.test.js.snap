// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleReportScreen tests renders correctly 1`] = `
<ScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "HelveticaNeue-Italic",
          "fontSize": 30,
        }
      }
    >
      Divano abbandonato
    </Text>
    <Text>
      9 Dicembre 2018 alle 17:40
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Avenir-Book",
          "fontSize": 14,
        }
      }
    >
      Divano abbandonato in via Gramsci
    </Text>
    <Text />
    <Text />
    <Component
      source={
        Object {
          "testUri": "../../../App/Assets/Images/default_report_image.png",
        }
      }
      style={
        Object {
          "height": 400,
          "width": 400,
        }
      }
    />
  </View>
</ScrollView>
`;
